arcgis-ios-sdk-samples/Scenes/Animate 3D graphic/CameraSettingsViewController.swift

```diff
@@ -16,141 +16,120 @@
 import UIKit
 import ArcGIS
 
-class CameraSettingsViewController: UIViewController {
+class CameraSettingsViewController: UITableViewController {
 
-    weak var orbitGeoElementCameraController:AGSOrbitGeoElementCameraController?
+    weak var orbitGeoElementCameraController: AGSOrbitGeoElementCameraController?
     
-    @IBOutlet var headingOffsetSlider:UISlider!
-    @IBOutlet var pitchOffsetSlider:UISlider!
-    @IBOutlet var distanceSlider:UISlider!
-    @IBOutlet var distanceLabel:UILabel!
-    @IBOutlet var headingOffsetLabel:UILabel!
-    @IBOutlet var pitchOffsetLabel:UILabel!
-    @IBOutlet var speedSlider:UISlider!
-    @IBOutlet var speedLabel:UILabel!
-    @IBOutlet var autoHeadingEnabledSwitch:UISwitch!
-    @IBOutlet var autoPitchEnabledSwitch:UISwitch!
-    @IBOutlet var autoRollEnabledSwitch:UISwitch!
+    @IBOutlet var headingOffsetSlider: UISlider!
+    @IBOutlet var pitchOffsetSlider: UISlider!
+    @IBOutlet var distanceSlider: UISlider!
+    @IBOutlet var distanceLabel: UILabel!
+    @IBOutlet var headingOffsetLabel: UILabel!
+    @IBOutlet var pitchOffsetLabel: UILabel!
+    @IBOutlet var autoHeadingEnabledSwitch: UISwitch!
+    @IBOutlet var autoPitchEnabledSwitch: UISwitch!
+    @IBOutlet var autoRollEnabledSwitch: UISwitch!
+    
+    private var distanceObservation: NSKeyValueObservation?
+    private var headingObservation: NSKeyValueObservation?
+    private var pitchObservation: NSKeyValueObservation?
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        //apply initial values to controls
-        self.setInitialValues()
+        guard let cameraController = orbitGeoElementCameraController else {
+            return
+        }
         
-        //add observers to update the sliders
-        self.orbitGeoElementCameraController?.addObserver(self, forKeyPath: #keyPath(AGSOrbitGeoElementCameraController.cameraDistance), options: .new, context: nil)
-        self.orbitGeoElementCameraController?.addObserver(self, forKeyPath: #keyPath(AGSOrbitGeoElementCameraController.cameraHeadingOffset), options: .new, context: nil)
-        self.orbitGeoElementCameraController?.addObserver(self, forKeyPath: #keyPath(AGSOrbitGeoElementCameraController.cameraPitchOffset), options: .new, context: nil)
+        // apply initial values to controls
+        updateUIForDistance()
+        updateUIForHeadingOffset()
+        updateUIForPitchOffset()
+        autoHeadingEnabledSwitch.isOn = cameraController.isAutoHeadingEnabled
+        autoPitchEnabledSwitch.isOn = cameraController.isAutoPitchEnabled
+        autoRollEnabledSwitch.isOn = cameraController.isAutoRollEnabled
+        
+        // add observers to the values we want to show in the UI
+        distanceObservation = cameraController.observe(\.cameraDistance) {[weak self] (controller, change) in
+            self?.updateUIForDistance()
```

Comment:
The change may not happen on the main thread, so the UI work needs to be dispatched there first:

```suggestion
            DispatchQueue.main.async {
                self?.updateUIForDistance()
            }
```